hetmech/hetmat/tests/test_archive.py

```diff
@@ -0,0 +1,42 @@
+import filecmp
+import pathlib
+import zipfile
+
+import hetmech.hetmat
+import hetmech.hetmat.archive
+import hetmech.matrix
+from hetmech.tests.hetnets import get_graph
+
+
+def test_disease_gene_example_hetmat_archiving(tmpdir):
+    """
+    Test archiving the hetmat corresponding to the hetnet in Figure 2C at https://doi.org/crz8.
+    """
+    tmpdir = pathlib.Path(tmpdir)
+    graph = get_graph('disease-gene-example')
+    hetmat_0_dir = tmpdir.joinpath('disease-gene-example-0.hetmat')
+    hetmat = hetmech.hetmat.hetmat_from_graph(graph, hetmat_0_dir)
+
+    # Test creating archive
+    archive_path = hetmech.hetmat.archive.create_hetmat_archive(hetmat)
+    with zipfile.ZipFile(archive_path) as zip_file:
+        name_list = zip_file.namelist()
+    expected = [
+        'edges/DlT.sparse.npz',
+        'edges/GaD.sparse.npz',
+        'edges/GeT.sparse.npz',
+        'edges/GiG.sparse.npz',
+        'metagraph.json',
+        'nodes/Disease.tsv',
+        'nodes/Gene.tsv',
+        'nodes/Tissue.tsv',
+    ]
+    assert name_list == expected
```

Comment:
Are the names in a zip file always ordered the same way? If not, should we use sets or sorted lists to be safe?

Reply:
Yes we want to test that the files are sorted in a deterministic manner